### PR TITLE
ARROW-5329: [MATLAB] Add support for building MATLAB interface to Feather directly within MATLAB

### PR DESCRIPTION
**Pull request by rdmello**
The MATLAB interface to Feather currently uses CMake to build MEX files. It would be convenient to be able to build the MEX file used by the MATLAB interface to Feather directly using normal MATLAB commands.

This will also provide a workaround for MEX compilation issues when using older CMake versions with newer versions of MATLAB (R2018b and beyond).

The commits in the pull request add a new `build_support` folder to the MATLAB subfolder in the Arrow repository. This `build_support` folder contains two functions called `compile` and `test` that use the `ARROW_HOME` environment variable and the `mex` command to build the necessary MEX files.

For example: 

``` matlab
>> cd build_support
>> compile
...lots of output...
>> test
...successfully passed all tests...
```

Thanks to Mark (@mcafaro) for implementing this and working through the initial review in https://github.com/mathworks/arrow/pull/13

### CODE REVIEW

**Comment by wesm** [comment]
Thanks for your pull request. I encourage you to conduct your code reviews in the main patch queue so the community (i.e. people Watching apache/arrow) can see them. We already have a very large PR queue so there is no concern about creating noise

File: matlab/build_support/test.m [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+% KIND, either express or implied.  See the License for the
+% specific language governing permissions and limitations
+% under the License.
+
+vars = common_vars();
+
+compile();
+
+originalPath = addpath(vars.srcDir, vars.buildDir);
+restoreOriginalPath = onCleanup(@()path(originalPath));
```
Comment (wesm) [inline matlab/build_support/test.m]: Is there a styleguide for MATLAB library code?

**Review by wesm — APPROVED** [review]
+1

**Comment by wesm** [comment]
@mcafaro @rdmello what is Mark's Apache JIRA id so I can assign this to him?

**Comment by mcafaro** [comment]
> @mcafaro @rdmello what is Mark's Apache JIRA id so I can assign this to him?

Hi @wesm, my Apache JIRA id is mcafaro. Thanks.

**Comment by rdmello** [comment]
Hello @wesm, thanks for merging this in. My apologies for not doing this as a code review in the main Apache Arrow pull request queue.

We'll be sure to do this for all future pull requests. However there is [one more pull request](https://github.com/mathworks/arrow/pull/6) on `mathworks/arrow` that Kevin's wrapping up right now. This started before we realized that it would be better to do draft pull requests directly on the main `apache/arrow` repository.

**Comment by wesm** [comment]
No problem. It's common that new contributors are unfamiliar with the Apache Way (https://apache.org/theapacheway/) so I'm here to help =)

File: matlab/build_support/test.m (hunk 1 above)
Comment (hatemhelal) [reply matlab/build_support/test.m]: Somewhat surprisingly, there isn't an official MATLAB style guide.  The closest one I'm aware of is [MATLAB Style Guidelines 2.0](https://www.mathworks.com/matlabcentral/fileexchange/46056-matlab-style-guidelines-2-0).

**Pull request closed without merging**